src/circuit_builder.rs

```diff
@@ -38,44 +273,162 @@ impl PublicInput {
 
 /// Circuit representation for a gadget with all of the tools that it
 /// should implement.
-pub trait Circuit<'a>
-where
-    Self: Sized + Default,
-{
+pub trait Circuit: Sized {
     /// Gadget implementation used to fill the composer.
-    fn gadget(&mut self, composer: &mut StandardComposer) -> Result<Vec<PublicInput>>;
-    /// Compiles the circuit by using a function that returns a `Result`
-    /// with the `ProverKey`, `VerifierKey` and the circuit size.
-    fn compile(&mut self, pub_params: &PublicParameters)
-        -> Result<(ProverKey, VerifierKey, usize)>;
+    fn gadget(&mut self, composer: &mut StandardComposer) -> Result<Vec<PublicInput>, Error>;
```

Comment:
Why do you need to add `Error` here? `anyhow::Result` already considers that.

Reply:
Fixed